hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -763,7 +738,7 @@ public long heapSize() {
   /**
    * @return true to indicate the block is allocated from JVM heap, otherwise from off-heap.
    */
-  boolean isOnHeap() {
+  public boolean isOnHeap() {
     return buf.hasArray();
```

Comment:
Nit : Just return 'false' here? Since you have HeapHfileBlock now where isOnheap() is always true?

Reply:
Yes, can be.

Reply:
Think about this again, lots of UT use the HFileBlock with heap buf , if just return false here, I guess those UT will be failure.... so plan to keep this.

Reply:
Thought about this before,  our ByteBuffAllocator won't allocate pooled heap BB , also if allocate a offheap BB, then it must be a pooled one.  That's to say, heapBB = exclusiveBB, offheapBB = sharedBB.  So if it's a heapBB, then must be a exclusiveBB..

Reply:
>Think about this again, lots of UT use the HFileBlock with heap buf , if just return false here, I guess those UT will be failure.... so plan to keep this.
Then why cant you just do the reverse way?  HFileBlock will be on heap one. So the UTs also dont need any change.  We will have an extension to it. Name it like SharedMemHFileBlock or OffheapHFileBlock (I prefer the former).  This will look a bit strange here where the HFileBlock as such is not telling whether it is on heap or off heap and we have a On heap specific extended class again!